src/services/shims.ts

```diff
@@ -351,8 +355,18 @@ namespace ts {
         constructor(private shimHost: CoreServicesShimHost) {
         }
 
-        public readDirectory(rootDir: string, extension: string): string[] {
-            var encoded = this.shimHost.readDirectory(rootDir, extension);
+        public readDirectory(rootDir: string, extension: string, exclude: string[]): string[] {
+            // Wrap the API changes for 1.5 release. This try/catch
+            // should be removed once TypeScript 1.5 has shipped.
+            // Also consider removing the optional designation for
+            // the exclude param at this time.
+            var encoded: string;
+            try {

```

Comment:
If we pass too many arguments, and there is a CoreServicesHost mismatch,  it will throw. We then fallback to the old call without the JSON encoded exclude string array.  (This is a similar to the fallback we already have in place for getDefaultLibFileName)